File: 0383-ransom-note/0383-ransom-note.go
func canConstruct(ransomNote string, magazine string) bool {
    alphaMap := make([]int, 26)
    for i := 0; i < len(magazine); i++ {
        alphaMap[magazine[i] - 'a']++
    }

    for i := 0; i < len(ransomNote); i++ {
        alphaMap[ransomNote[i] - 'a']--
        if alphaMap[ransomNote[i] - 'a'] < 0 {
            return false
        }
    }
    
    return true
}

File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int alpha[26] = {0};
        for (int i = 0; i < magazine.size(); i++) {
            alpha[magazine[i] - 'a']++;
        }

        //for (int i = 0; i < 26; i++) {
        //    cout << i << " " << alpha[i] << endl; 
        //}

        for (int i = 0; i < ransomNote.size(); i++) {
            int idx = ransomNote[i] - 'a';
            if (alpha[idx] == 0) {
                return false;
            }

            alpha[idx]--;
        }

        return true;
    }
};